geoh5py/shared/utils.py

```diff
@@ -382,6 +382,34 @@ def dict_mapper(
     return val
 
 
+def box_intersect(
+    extent_a: np.ndarray, extent_b: np.ndarray, return_array=False
+) -> bool | np.ndarray:
+    """
+    Compute the intersection of two axis-aligned bounding extents defined by their
+    bottom south-west corner and top north-east corner.
+
+    :param extent_a: First extent
+    :param extent_b: Second extent
+    :param return_array: Return the intersection extent
+
+    :return: True if the box extents intersect,
+        or intersection extent coordinates if `return_array` is True.
+    """
+    min_ext = []
+    max_ext = []
+    for comp_a, comp_b in zip(extent_a.T, extent_b.T):
```

Comment:
make it complciate becaseu there is no axumtion that it point are ordered (min_x, min_y) (max_x, max_y)
Could just assert if it is no the case. Would also make less test cases to write...

Extent should always be correctly ordered when computed. If user provided, it is the responsibility of the entry point, in upstream code, to fixup the box or raise an exception

Reply:
Correct, will add a check on order.